### PR TITLE
[2.1] Fallback to the dummy audio driver if no other audio driver works

### PR DESCRIPTION
**Pull request by xsellier**
Fix this issue https://github.com/godotengine/godot/issues/1684
Cannot cherry-pick those [changes](https://github.com/godotengine/godot/pull/11252), had to make the fix manually

### CODE REVIEW

File: platform/javascript/os_javascript.cpp [hunk 1]
```diff
@@ -440,12 +440,7 @@ void OS_JavaScript::initialize(const VideoMode &p_desired, int p_video_driver, i
 	visual_server = memnew(VisualServerRaster(rasterizer));
 	visual_server->init();
 
-	/*AudioDriverManagerSW::get_driver(p_audio_driver)->set_singleton();
-
-	if (AudioDriverManagerSW::get_driver(p_audio_driver)->init()!=OK) {
-
-		ERR_PRINT("Initializing audio failed.");
-	}*/
+	AudioDriverManagerSW::initialize(p_audio_driver);
```
Comment (akien-mga) [inline platform/javascript/os_javascript.cpp]: Does it work? If those were commented out I assume it means that audio was not working in 2.1 JavaScript?

File: platform/windows/os_windows.cpp [hunk 2]
```diff
@@ -2621,6 +2616,7 @@ OS_Windows::OS_Windows(HINSTANCE _hInstance) {
 #ifdef RTAUDIO_ENABLED
 	AudioDriverManagerSW::add_driver(&driver_rtaudio);
 #endif
+	AudioDriverManagerSW::add_driver(&driver_dummy);
```
Comment (akien-mga) [inline platform/windows/os_windows.cpp]: Why does Windows get this added specifically when other platforms get it removed? It seems `AudioDriverManagerSW::initialize` already takes care of adding a dummy fallback.

File: platform/javascript/os_javascript.cpp (hunk 1 above)
Comment (xsellier) [reply platform/javascript/os_javascript.cpp]: Forgot to comment it out after updating it :-1:

File: platform/windows/os_windows.cpp (hunk 2 above)
Comment (xsellier) [reply platform/windows/os_windows.cpp]: Done

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**